recipe/build-mpi.sh

```diff
@@ -1,32 +1,36 @@
 #!/bin/bash
 
-# unset unused old fortran compiler vars
-unset F90 F77
-
 set -ex
 
-export FCFLAGS="$FFLAGS"
-
 # avoid absolute-paths in compilers
 export CC=$(basename "$CC")
 export CXX=$(basename "$CXX")
 export FC=$(basename "$FC")
 
-./autogen.pl --force
+# unset unused Fortran compiler variables
+unset FFLAGS F77 F90 F95
 
+# tweak compiler flags
+export LIBRARY_PATH="$PREFIX/lib"
 if [[ "$target_platform" == osx-* ]]; then
-    if [[ ! -z "$CONDA_BUILD_SYSROOT" ]]; then
+    if [[ -n "$CONDA_BUILD_SYSROOT" ]]; then
         export CFLAGS="$CFLAGS -isysroot $CONDA_BUILD_SYSROOT"
         export CXXFLAGS="$CXXFLAGS -isysroot $CONDA_BUILD_SYSROOT"
     fi
 fi
 
-if [[ -z $CUDA_HOME ]]; then
-    build_with_cuda=""
-else
+# UCX support
+build_with_ucx=""
+if [[ "$target_platform" == linux-* ]]; then
+    build_with_ucx="--with-ucx=$PREFIX"
+fi
+
+# CUDA support
+build_with_cuda=""
+if [[ -n "$CUDA_HOME" ]]; then
+    build_with_cuda="--with-cuda"
     export CFLAGS="$CFLAGS -I$CUDA_HOME/include"
     export CXXFLAGS="$CXXFLAGS -I$CUDA_HOME/include"
```

Comment:
Note: This is OK if we still build with CUDA 11 (thus me asking Jeff about the CUDA support above). For CUDA 12+, there's no more `$CUDA_HOME` as we use plain docker images without pre-installed CUDA.